### ISSUE TITLE
{{noteContent}} not working in every file format

### ISSUE THREAD

**Issue by ScriBanana** [argenos/zotero-mdnotes#44]
(I posted this in #43 as comments, but I am rewriting it properly here for the sake of clarity.)

I can't figure out how to configure a template to include my highlights and annotations. I use the single file approach, with some metadata, links and annotations and I didn't manage to include the later in my MD files.

It seems that `{{noteContent}}` only works in the files that use the `Zotero Note Template.md` format. And as far as I know, there is no command to export that single file in the right-click menu, with the _single file_ organisation.
It is included in the batch export, but I don't think we have its template available to document what it does.

**Comment by argenos** [comment]
Hi @ScriBanana, `{{noteContent}}` indeed only works with the Zotero Note template (see [here](https://argentinaos.com/zotero-mdnotes/docs/placeholders/#note-placeholders), the main reason is that note content refers to the particular note you're trying to export. In a single file setting, with an item that has more than one note that would be a bit ambiguous, i.e. the content of which note? 

I must say that I have been tempted to remove the *single file* option now that templates are available, as it makes this a lot more confusing, but I haven't really understood how it's being used by other people.  With the single file option, do you only want `Export to markdown` that includes the metadata and note contents? 

For now, if you want all contents in a single file, you should use the `Batch export to markdown`. You should probably edit the other templates (particularly the default mdnotes template) to remove duplicate fields.

**Comment by ScriBanana** [comment]
Thank you.

"_Note content refers to the particular note you're trying to export [etc]_" That makes sense, I understand the issue. I used to export everything _via_ `batch export` before the update, that's why I didn't even understand that there was a single file for each note.
I shall make a default template with what I want, and empty the other templates before the notes to have a clean file for each reference after a batch export.

Well I personally would be rather annoyed if you dropped the _single file_ option! :)
My work process consists in making notes on single concepts, that include ideas from specific papers. These notes are linked to each paper single MD file, which contains all the essential info I might need to remember what it is about and what I got from it (a few metadata, sometimes the abstract, some citations and my notes).
So yes, my wish is to get a command to generate a single note with the metadata I choose in the style I choose with my annotations, pdf notes and additional notes I write in Obsidian afterwards (bad habit, I know). Until now, I would do it with a `batch export`, but the lack of format customisation was a bit annoying. So the template feature now makes the wish come true. (Despite my initial incapacity to properly understand how Mdnotes work)

**Comment by argenos** [comment]
@ScriBanana not to worry! I didn't remove it because I assumed some workflows were relying on it. I would rather adapt the current use of templates to fit that workflow too before I end up removing anything. (it's not just you, this setting is confusing for others and myself!)

I have a question though, does this *single-file* setting behave more like the mdnotes file? i.e. you don't want it replaced since it contains your notes? The current `Batch export to markdown` **will** replace the file, so it is indeed a bit dangerous. 

I think one solution would be to add a placeholder that lets you add the formatted notes and/or a template for a single file.

**Comment by ScriBanana** [comment]
I am not certain I understand what "_the mdnotes file_" refers to. Do you mean that I would rather have `batch export` ask me before erasing a file the way the other commands do? Well yes, it would be nice if it was an option. I made an "annotated" Zotero tag to prevent me from accidentally deleting my note through an unfortunate export, but the risk still exists. (I set up automatic backups in case I fail to pay attention to it).

A template made for single file export would likely be the simplest option for a user like me, but once one understands how single file batch export works, it is not too much of a hassle to set up properly the way we discussed above (I guess, I didn't have time to try it since my last messages).
I don't get what the placeholder you propose would do. "Notes" is a bit too polysemous in our context. :)

The holy grail would be having the possibility to update already written MD files, e.g. after adding a tag or modifying metadata in Zotero, without affecting the part added to the file on the MD reader software. With a comparison between the placeholder of the template and the content of the already existing note, for example. But that would be of course be a whole programming project with many uncertainties related to each user, so it is likely vain. Also, due to the diversity of PDF formats and uses, it is common to have to rework extracted annotations, like removing line numbers on submitted papers, for example, so there is always the need for a bit of fiddling with each note anyway.

On a side-note, it is interesting to me that you find the single file approach confusing. I would expect most people not to want several MD notes per paper, with information scattered all around the place, and yet it doesn't seem to be common among Mdnotes users. Maybe I misunderstood the purpose of the markdown format. :)

**Comment by ScriBanana** [comment]
Hello,
Small update; I finally found the time to setup the batch export with edited templates. It works well!

It raised a little suggestion, but not enough that it would deserve a full issue. I edited the title placeholder to remove the `# ` and added it to the templates, so that I could get different levels of stress for each part of my single file (that is, for each type of file for you). Basically, the standard template starts with `# {{title}}` and the notes template with `## {{title}}`, which makes more sense in my single files. I wonder if it wouldn't be clearer to transfer formatting from placeholders to the templates, since not everyone might want a single `#` for their title depending on their CSS, for example, and because editing a template is more straightforward than it is a placeholder. It is of course not possible for elements that are listed.

**Comment by argenos** [comment]
> I wonder if it wouldn't be clearer to transfer formatting from placeholders to the templates, since not everyone might want a single # for their title depending on their CSS, for example, and because editing a template is more straightforward than it is a placeholder. It is of course not possible for elements that are listed.

Fields that are lists are exactly what lead me to the placeholder idea! The nice thing with the current solution is that it's flexible enough to allow the users to decide where to place the formatting for non-list fields (e.g., in this case, the headings in the template vs the placeholder). I don't think I'll move this to the templates, but I'll definitely include this in the documentation! 

Something to keep in mind with the placeholders: they also help with "removing" fields that are empty or not available in the current item. The title is something all the items will have, but, for other fields, adding the formatting in the template can lead to orphan symbols.
As an example, if you have a template like this:
```markdown
{{proceedingsTitle}}
**{{publicationTitle}}**
```
And you export a conference article (which has an empty `publicationTitle` field), you'll get something like this:
```markdown
Proceedings title: My awesome proceedings
****
```

@brimwats come to think of it, this is also a solution for #47

**Comment by ScriBanana** [comment]
Hmmm yes, that makes sense. My issue raised from the fact that it is a bit finickier to modify a placeholder and to understand their fields and allowed values than adding the wanted symbols to the template. For example, I find easier to understand that one can add `[[ ]]` around `{{tags}}` in a template than it is to find out that `"link_style": "wiki"` is a thing and how to add it to the placeholder. Of course, the brackets will suffice in the placeholder, but still, I don't find them simple to use, maybe because of Zotero's GUI.
I reckon some users (at least me) are not used to fiddling with Zotero placeholders when they get their hands on your program, and getting into Mdnotes template modification is likely easier. But yes, this is a bit stuck, as you said.

**Comment by argenos** [comment]
The documentation for this should be fixed for v0.1.3
